api.go

```diff
@@ -23,6 +23,9 @@ import (
     `github.com/bytedance/sonic/internal/rt`
 )
 
+// APIKind is the kind of API, 0 is std json, 1 is sonic.
+const APIKind = apiKind
```

Comment:
maybe we need use more friendly name, such as
```
 type APIKind uint64
 const (
        UseSonicJson = iota
        UseStdJson
 )
```

Reply:
My initial idea was the same as yours, but considering the dependency relationship between files, api.go should depend on compact.go and sonic.go. If we write such code, we will break this dependency relationship.

Reply:
> maybe we need use more friendly name, such as
> ```
>  type APIKind uint64
>  const (
>         UseSonicJson = iota
>         UseStdJson
>  )
> ```

My initial idea was the same as yours, but considering the dependency relationship between files, api.go should depend on compact.go and sonic.go. If we write such code, we will break this dependency relationship.

But we do need more user-friendly code, and I will make changes.

Reply:
I’ve made some improvements, please review again.